in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,265 @@
+  object, and path to the key file. Also there are two optional inputs
+  based on which it decides whether to sign the file with the replacement
+  of the existing signature, or sign it but without replacement and then
+  appends the new signature in the file. Further in each of the two cases,
+  depending on the arguments it then dumps the file either by infixing
+  the keyID i.e. <filename>.<8-initial-characters-from-the-keyID>.link
+  or simpy <filename>.link
+
+  General Usage:
+  python in_toto_sign.py --key <path/to/key> <sign/verify> [-r] [-i]
+    <path/to/link/file>
+
+  Example Usage:
+  Suppose Bob wants to sign a file called package.link, and also while signing,
+  he wants to replace all the existing signatures, and then dump the file
+  by infixing the keyID in it. Then his command would be-
+
+  python in_toto_sign.py --key /bob/mykeys/bob_pvt_key
+    sign -r -i  /bob/software/in-toto/test/package.link
+
+"""
+import os
+import sys
+import argparse
+import in_toto.log
+import in_toto.util
+from in_toto.models.common import Signable as signable_object
+from in_toto.models.link import Link as link_import
+import securesystemslib.exceptions
+import securesystemslib.keys
+import securesystemslib.formats
+
+
+def add_sign(link, key):
+    """
+    <Purpose>
+      Signs the given link file with the corresponding key,
+      adds the signature to the file, and then returns, the
+      link file as an object.
+
+    <Arguments>
+      link - path to the signable link file
+      key - the key to be used for signing
+
+    <Exceptions>
+      None
+
+    <Returns>
+      An object containing the contents of the link file after
+      adding the signature
+
+    """
+    # reading the file from the given link and making an object
+    signable_object = link_import.read_from_file(link)
+
+    rsa_key = in_toto.util.import_rsa_key_from_file(key)
+
+    # checking if the given key follows the format
+    securesystemslib.formats.KEY_SCHEMA.check_match(rsa_key)
+
+    # create the signature using the key, and the link file
+    signature = securesystemslib.keys.create_signature(rsa_key, signable_object.payload)
+
+    # append the signature into the file
+    signable_object.signatures.append(signature)
+
+    return signable_object
+
+
+def add_replace(link, key):
+    """
+    <Purpose>
+      Replaces all the exisiting signature with the new signature,
+      signs the file, and then returns the link file as an object.
+
+    <Arguments>
+      link - path to the key file
+      key - the key to be used for signing
+
+    <Exceptions>
+      None
+
+    <Returns>
+      An object containing the contents of the link file after
+      adding the signature which replaces the old signatures
+    """
+
+    # reading the link file and making an object
+    signable_object = link_import.read_from_file(link)
+
+    # import rsa key from the filepath
+    rsa_key = in_toto.util.import_rsa_key_from_file(key)
```

Comment:
This won't work if the private key is encrypted. You can use `util.prompt_import_rsa_key_from_file` instead. Also below.